### PR TITLE
Fixes #21854: Properly configure CSP violation handler

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/21854

### CODE REVIEW

**Comment by amousset** [comment]
Produces logs like:

```
WARN  application - Content security policy violation: blocked inline in https://127.0.0.1:8181/rudder/ because of script-src-elem directive
```

**Comment by amousset** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala [hunk 1]
```diff
@@ -362,6 +364,12 @@ class Boot extends Loggable {
     // the session itself.
     LiftRules.noCometSessionCmd.default.set(() => JsCmd.unitToJsCmd(()))
 
+    // Log CSP violations
+    LiftRules.contentSecurityPolicyViolationReport = (r: ContentSecurityPolicyViolation) => {
+      ApplicationLogger.warn(s"Content security policy violation: blocked ${r.blockedUri} in ${r.documentUri} because of ${r.violatedDirective} directive")
```
Comment (amousset) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala:369]: I haven't investigated how to log is the right context, Application looks a bit too generic.

Comment (fanf) [reply webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala:369]: Seems correct to me

**Review by fanf — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**